featurebyte/service/feature_manager.py

```diff
@@ -202,46 +227,157 @@ async def _populate_feature_store(
         schedule_time: datetime,
         aggregation_result_name: str,
     ) -> None:
-        next_job_time = get_next_job_datetime(
+        job_schedule_ts = get_previous_job_datetime(
             input_dt=schedule_time,
             frequency_minutes=tile_spec.frequency_minute,
             time_modulo_frequency_seconds=tile_spec.time_modulo_frequency_second,
         )
-        job_schedule_ts = next_job_time - timedelta(minutes=tile_spec.frequency_minute)
         job_schedule_ts_str = job_schedule_ts.strftime("%Y-%m-%d %H:%M:%S")
         await self.tile_manager_service.populate_feature_store(
             session, tile_spec, job_schedule_ts_str, aggregation_result_name
         )
 
-    async def _generate_historical_tiles(self, session: BaseSession, tile_spec: TileSpec) -> None:
-        # generate historical tile_values
-        date_format = "%Y-%m-%dT%H:%M:%S.%fZ"
+    async def _backfill_tiles(
+        self, session: BaseSession, tile_spec: TileSpec, schedule_time: datetime
+    ) -> None:
+        """
+        Backfill tiles required to populate internal online store
 
-        # derive the latest tile_start_date
-        end_ind = date_util.timestamp_utc_to_tile_index(
-            datetime.utcnow(),
-            tile_spec.time_modulo_frequency_second,
-            tile_spec.blind_spot_second,
-            tile_spec.frequency_minute,
-        )
-        end_ts = date_util.tile_index_to_timestamp_utc(
-            end_ind,
-            tile_spec.time_modulo_frequency_second,
-            tile_spec.blind_spot_second,
-            tile_spec.frequency_minute,
-        )
-        end_ts_str = end_ts.strftime(date_format)
+        This will determine the tiles that need to be backfilled using information stored in the
+        tile registry collection. Tiles between backfill_start_date and last_run_tile_end_date can
+        be assumed to be available because of previous backfills and on going scheduled tile tasks.
+        See an example below.
+
+        Feature: Amount Sum over 48 hours
+
+                 <-------------------------------- 48h ------------------------------->
+                                            (tiles available)
+                                      # # # # # # # # # # # # # # #
+        ---------|-------------------|----------------------------|-------------------|--> (time)
+                 ^                   ^                            ^                   ^
+             start_ts        backfill_start_date      last_run_tile_end_date       end_ts
+                 <------------------->                            <------------------->
+                    required compute                                 required compute
```

Comment:
It is possible that `start_ts` and `end_ts` interval partially overlaps with `backfill_start_date` and `last_run_tile_end_date` interval?

Reply:
That is possible. When that occurs, some tiles computation can be skipped since the tiles are already available. For example, if `start_ts` is after `backfill_start_date`, the logic will skip calculating the tiles between that interval.